### PR TITLE
Feat/data link structure

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |y|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

# Fixes
- ems:contenttype:clean field type is unset
- wysiwyg fieldType language not correct passed 
  > NL_be should be passed as NL
- wysiwyg internal link send language
  > The language should be passed when creating a ems link from a wysiwyg (ckeditor)

# Features
- new contentType view DataLink
> - Add ViewTypes a collection object for all available contentType views
> - Remove ViewTypeCompilerPass.php, replaced by new collection ViewTypes
> - Add view_types.xml
> - Add helper methods on ContentType (giveLabel, hasLabel, getFirstViewByType, ...)
> - Add a new view type "ems.view.data_link"
> - Menu don't display views of the new viewType "ems.view.data_link"
> - Refactor searchApiAction too work with new object DataLinks instead of search.json.twig:
> - If searching on 1 contentType and the contentType has a view of the type dataLink use this view to fill the DataLinks object.

### CODE REVIEW

File: src/Entity/ContentType.php [hunk 1]
```diff
@@ -1043,6 +1043,11 @@ public function getExtra()
         return $this->extra;
     }
 
+    public function hasFieldType(): bool
+    {
+        return null !== $this->fieldType;
+    }
+
```
Comment (dameert) [inline src/Entity/ContentType.php:1050]: I like this approach!

File: doc/contentTypes/views.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+# Content Type: Views
+
+Name | description 
+--- | ---
+[CalendarViewType](#calendar-view) | A view where you can schedule your object  
+[CriteriaViewType](#criteria-view) | A view where we can massively content types having criteria
```
Comment (dameert) [inline doc/contentTypes/views.md:6]: massively edit content

File: doc/contentTypes/views.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+[KeywordsViewType](#keywords-view) | A view where all properties of kind (such as keyword) are listed on a single page 
+[ReportViewType](#report-view) | Perform an elasticsearch query and generate a report with a twig template 
+[SorterViewType](#sorter-view) | Order a sub set (based on a ES query) 
+ 
+
+## Calendar view
+A view where you can schedule your object
+
+## Criteria view
+A view where we can massively content types having criteria
```
Comment (dameert) [inline doc/contentTypes/views.md:21]: massively edit content

File: src/Form/View/CriteriaViewType.php [hunk 4]
```diff
@@ -29,7 +29,7 @@ public function __construct(FormFactory $formFactory, Environment $twig, LoggerI
 
     public function getLabel(): string
     {
-        return 'Criteria: a view where we can massively content types having critetira';
+        return 'Criteria: a view where we can massively content types having criteria';
```
Comment (dameert) [inline src/Form/View/CriteriaViewType.php:32]: can massively edit content

**Review by dameert — APPROVED** [review]

File: src/Form/View/CriteriaViewType.php (hunk 4 above)
Comment (Davidmattei) [reply src/Form/View/CriteriaViewType.php]: c7c590d

File: doc/contentTypes/views.md (hunk 3 above)
Comment (Davidmattei) [reply doc/contentTypes/views.md]: c7c590d

File: doc/contentTypes/views.md (hunk 2 above)
Comment (Davidmattei) [reply doc/contentTypes/views.md]: c7c590d

**Review by dameert — APPROVED** [review]

**Review by theus77 — APPROVED** [review]

**Pull request merged**